tools/noopfactory/src/main/kotlin/com/datadog/tools/noopfactory/NoOpFactoryAnnotationProcessor.kt

```diff
@@ -87,13 +88,24 @@ class NoOpFactoryAnnotationProcessor : AbstractProcessor() {
         val packageName = typeElement.qualifiedName.toString().substringBeforeLast('.')
         val typeSpec = generateTypeSpec(typeElement, packageName)
 
-        val file = FileSpec.builder(packageName, typeSpec.name.orEmpty())
+        val className = typeSpec.name.orEmpty()
+        val file = FileSpec.builder(packageName, className)
             .addType(typeSpec)
+            .indent("    ")
             .build()
 
-        file.writeTo(File(kaptKotlinGeneratedDir))
+        val kotlinFileObject = processingEnv.filer
+            .createResource(
+                StandardLocation.SOURCE_OUTPUT,
+                packageName,
+                "$className.kt",
+                typeElement
+            )
+        val writer = kotlinFileObject.openWriter()
```

Comment:
I am wondering if we should use a try/finally here to make sure this is close ?

Reply:
In theory those writer methods should be safe.